src/releases/DepartureReleaseEventHandler.cpp

```diff
@@ -0,0 +1,909 @@
+            TaskManager::TaskRunnerInterface& taskRunner,
+            Euroscope::EuroscopePluginLoopbackInterface& plugin,
+            const Controller::ControllerPositionCollection& controllers,
+            const Controller::ActiveCallsignCollection& activeCallsigns,
+            const Dialog::DialogManager& dialogManager,
+            const int triggerRequestDialogFunctionId,
+            const int triggerDecisionMenuFunctionId,
+            const int releaseDecisionCallbackId,
+            int releaseCancellationCallbackId
+        ): triggerRequestDialogFunctionId(triggerRequestDialogFunctionId),
+           triggerDecisionMenuFunctionId(triggerDecisionMenuFunctionId),
+           releaseDecisionCallbackId(releaseDecisionCallbackId),
+           controllers(controllers), plugin(plugin), dialogManager(dialogManager), api(api), taskRunner(taskRunner),
+           activeCallsigns(activeCallsigns), releaseCancellationCallbackId(releaseCancellationCallbackId)
+        { }
+
+        void DepartureReleaseEventHandler::ProcessWebsocketMessage(const Websocket::WebsocketMessage& message)
+        {
+            if (message.event == "departure_release.requested") {
+                this->ProcessDepartureReleaseRequestedMessage(message.data);
+            } else if (message.event == "departure_release.approved") {
+                this->ProcessRequestApprovedMessage(message.data);
+            } else if (message.event == "departure_release.rejected") {
+                this->ProcessRequestRejectedMessage(message.data);
+            } else if (message.event == "departure_release.acknowledged") {
+                this->ProcessRequestAcknowledgedMessage(message.data);
+            } else if (message.event == "departure_release.request_cancelled") {
+                this->ProcessRequestCancelledMessage(message.data);
+            }
+        }
+
+        std::set<Websocket::WebsocketSubscription> DepartureReleaseEventHandler::GetSubscriptions() const
+        {
+            return {
+                {
+                    Websocket::WebsocketSubscription::SUB_TYPE_CHANNEL,
+                    "private-departure-releases"
+                }
+            };
+        }
+
+        void DepartureReleaseEventHandler::AddReleaseRequest(std::shared_ptr<DepartureReleaseRequest> request)
+        {
+            std::lock_guard queueLock(this->releaseMapGuard);
+            this->releaseRequests[request->Id()] = request;
+        }
+
+        std::shared_ptr<DepartureReleaseRequest> DepartureReleaseEventHandler::GetReleaseRequest(int id)
+        {
+            std::lock_guard queueLock(this->releaseMapGuard);
+            auto request = this->releaseRequests.find(id);
+            return request == this->releaseRequests.cend() ? nullptr : request->second;
+        }
+
+        bool DepartureReleaseEventHandler::DepartureReleaseRequestedMessageValid(const nlohmann::json& data) const
+        {
+            return data.is_object() &&
```

Comment:
This might be slightly more readable if you kind of split it into:#
- contains checks
- type checks
And then just return an and on those two?